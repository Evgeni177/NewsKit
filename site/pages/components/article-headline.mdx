import Layout from '../../components/layout';
import Playground from '../../components/playground';
import {Prop} from '../../components/prop';
import {TextBoxSwatch} from '../../components/theming-values/colors';
import {LegacyBlock} from '../../components/legacy-block';

import {ArticleHeadline} from 'newskit';

export default Layout;

# Article Headline

## Overview

### Usage

No special requirements.

### Theming

## Props

<LegacyBlock marginTop="sizing040">
  <Prop name="children" type="string" required>
    The content of the article headline is passed as the child of the component.
  </Prop>
  <Prop name="kickerText" type="string">
    The optional kicker content, which appears before the headline content.
  </Prop>
  <Prop name="kickerColor" type="ColorKeys">
    The optional color of the article kicker. If passed without kickerText, this prop has no effect.
  </Prop>
</LegacyBlock>

## Playground

<Playground
  componentName="ArticleHeadline"
  newskitPath="/article-headline"
  component={ArticleHeadline}
  knobs={[
    {
      name: 'Content',
      propName: 'children',
      value: 'ArticleHeadline Content',
    },
    {
      name: 'Kicker Text',
      propName: 'kickerText',
      value: 'ArticleKicker',
    },
    {
      name: 'Kicker Color',
      propName: 'kickerColor',
      options: [
        {
          label: 'inkContrast',
          value: 'inkContrast',
          isDefault: true,
        },
        {
          label: 'interactive010',
          value: 'interactive010',
        },
        {
          label: 'semanticNegative',
          value: 'semanticNegative',
        },
        {
          label: 'semanticPositive',
          value: 'semanticPositive',
        },
      ],
    },
  ]}
/>

## Rationale

### Styling

### Accessibility

#### Colour

### SEO

#### Ranking

#### Navigation

#### Descriptive
