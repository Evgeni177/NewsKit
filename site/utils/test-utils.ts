import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
