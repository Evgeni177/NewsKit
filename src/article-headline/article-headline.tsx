import React from 'react';
import {H1} from '../typography';
import {
  styled,
  getDefaultTypePreset,
  getDefaultMarginPreset,
  MQ,
} from '../utils/style';
import {getDefaultStylePreset} from '../utils/style-preset';
import {TypePresetKeys} from '../themes/mappers/type-presets';
import {MarginPresetKeys} from '../themes/mappers/spacing';
import {StylePresetKeys} from '../themes/mappers/style-preset';

export interface ArticleHeadline {
  kickerText?: string;
  renderHeadingAs?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span';
  renderKickerAs?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span';
  kickerStylePreset?: MQ<StylePresetKeys>;
  headingStylePreset?: MQ<StylePresetKeys>;
  kickerMqTypePreset?: MQ<TypePresetKeys>;
  headingMqTypePreset?: MQ<TypePresetKeys>;
  kickerMqInlineMarginPreset?: MQ<MarginPresetKeys>;
}

interface HeadingProps {
  headingStylePreset?: MQ<StylePresetKeys>;
  headingMqTypePreset?: MQ<TypePresetKeys>;
  as?: React.ElementType;
}

interface KickerProps {
  kickerStylePreset?: MQ<StylePresetKeys>;
  kickerMqTypePreset?: MQ<TypePresetKeys>;
  kickerMqInlineMarginPreset?: MQ<MarginPresetKeys>;
}

const Headline = styled.section`
  display: block;
`;

const Heading = styled(H1)<HeadingProps>`
  display: inline-block;
  ${getDefaultTypePreset('articleHeadline.heading', 'headingMqTypePreset', {
    withCrop: true,
  })}
  ${getDefaultStylePreset('articleHeadline.heading', 'headingStylePreset')}
`;

const Kicker = styled(Heading)<KickerProps>`
  display: inline-block;
  ${getDefaultTypePreset('articleHeadline.kicker', 'kickerMqTypePreset', {
    withCrop: true,
  })}
  ${getDefaultStylePreset('articleHeadline.kicker', 'kickerStylePreset')}
  ${getDefaultMarginPreset(
    'articleHeadline.kicker',
    'kickerMqInlineMarginPreset',
  )}
  text-transform: uppercase;
`;

export const ArticleHeadline: React.FC<ArticleHeadline> = ({
  children,
  kickerText,
  renderHeadingAs = 'h1',
  renderKickerAs = 'span',
  headingStylePreset,
  kickerStylePreset,
  kickerMqTypePreset,
  headingMqTypePreset,
  kickerMqInlineMarginPreset,
}) =>
  kickerText ? (
    <Headline>
      <Kicker
        as={renderKickerAs}
        kickerStylePreset={kickerStylePreset}
        kickerMqTypePreset={kickerMqTypePreset}
        kickerMqInlineMarginPreset={kickerMqInlineMarginPreset}
      >
        {kickerText}{' '}
      </Kicker>
      <Heading
        as={renderHeadingAs}
        headingStylePreset={headingStylePreset}
        headingMqTypePreset={headingMqTypePreset}
      >
        {children}
      </Heading>
    </Headline>
  ) : (
    <Heading as={renderHeadingAs}>{children}</Heading>
  );
