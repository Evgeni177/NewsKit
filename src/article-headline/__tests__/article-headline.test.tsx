import {ArticleHeadline} from '../article-headline';
import {renderToFragmentWithTheme} from '../../test/test-utils';

describe('ArticleHeadline', () => {
  test('renders correctly with default props', () => {
    const fragment = renderToFragmentWithTheme(ArticleHeadline, {
      children: 'test',
    } as any) as any;

    expect(fragment).toMatchSnapshot();
  });

  test('renders correctly with default props', () => {
    const fragment = renderToFragmentWithTheme(ArticleHeadline, {
      children: 'test',
      kickerText: 'this is kicker',
      renderHeadingAs: 'h4',
      renderKickerAs: 'h5',
    } as any) as any;

    expect(fragment).toMatchSnapshot();
  });
});
