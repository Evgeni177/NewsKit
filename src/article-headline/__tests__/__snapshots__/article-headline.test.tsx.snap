// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeadline renders correctly with default props 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 40px;
  line-height: 1.1;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0a0a0a;
  margin: 0 0 24px 0;
  display: inline-block;
  font-family: "Noto Sans",sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0a0a0a;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.21250000000000002em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.275em;
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 32px;
    line-height: 1.125;
    font-weight: 700;
    -webkit-letter-spacing: -0.5px;
    -moz-letter-spacing: -0.5px;
    -ms-letter-spacing: -0.5px;
    letter-spacing: -0.5px;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.21250000000000002em;
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.275em;
  }
}

<h1
    class="emotion-0"
  >
    test
  </h1>
</DocumentFragment>
`;

exports[`ArticleHeadline renders correctly with default props 2`] = `
<DocumentFragment>
  .emotion-1 {
  font-family: "Noto Sans",sans-serif;
  font-size: 40px;
  line-height: 1.1;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0a0a0a;
  margin: 0 0 24px 0;
  display: inline-block;
  font-family: "Noto Sans",sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0a0a0a;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.21250000000000002em;
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.275em;
}

@media screen and (min-width:960px) {
  .emotion-1 {
    font-family: "Noto Sans",sans-serif;
    font-size: 32px;
    line-height: 1.125;
    font-weight: 700;
    -webkit-letter-spacing: -0.5px;
    -moz-letter-spacing: -0.5px;
    -ms-letter-spacing: -0.5px;
    letter-spacing: -0.5px;
  }

  .emotion-1::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.21250000000000002em;
  }

  .emotion-1::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.275em;
  }
}

.emotion-2 {
  display: block;
}

.emotion-0 {
  font-family: "Noto Sans",sans-serif;
  font-size: 40px;
  line-height: 1.1;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0a0a0a;
  margin: 0 0 24px 0;
  display: inline-block;
  font-family: "Noto Sans",sans-serif;
  font-size: 32px;
  line-height: 1.125;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0a0a0a;
  display: inline-block;
  font-family: "Noto Sans",sans-serif;
  font-size: 36px;
  line-height: 1.1111111111111112;
  font-weight: 700;
  -webkit-letter-spacing: -0.5px;
  -moz-letter-spacing: -0.5px;
  -ms-letter-spacing: -0.5px;
  letter-spacing: -0.5px;
  color: #0A68C1;
  margin: 0 12px 0 0;
  text-transform: uppercase;
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.21250000000000002em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.275em;
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 32px;
    line-height: 1.125;
    font-weight: 700;
    -webkit-letter-spacing: -0.5px;
    -moz-letter-spacing: -0.5px;
    -ms-letter-spacing: -0.5px;
    letter-spacing: -0.5px;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.21250000000000002em;
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.275em;
  }
}

.emotion-0::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.2055555555555556em;
}

.emotion-0::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.2680555555555556em;
}

@media screen and (min-width:480px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 36px;
    line-height: 1.1111111111111112;
    font-weight: 700;
    -webkit-letter-spacing: -0.5px;
    -moz-letter-spacing: -0.5px;
    -ms-letter-spacing: -0.5px;
    letter-spacing: -0.5px;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.2055555555555556em;
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.2680555555555556em;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 40px;
    line-height: 1.1;
    font-weight: 700;
    -webkit-letter-spacing: -0.5px;
    -moz-letter-spacing: -0.5px;
    -ms-letter-spacing: -0.5px;
    letter-spacing: -0.5px;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.20000000000000007em;
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.26250000000000007em;
  }
}

@media screen and (min-width:1440px) {
  .emotion-0 {
    font-family: "Noto Sans",sans-serif;
    font-size: 44px;
    line-height: 1.0909090909090908;
    font-weight: 700;
    -webkit-letter-spacing: -0.5px;
    -moz-letter-spacing: -0.5px;
    -ms-letter-spacing: -0.5px;
    letter-spacing: -0.5px;
  }

  .emotion-0::before {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-bottom: -0.19545454545454544em;
  }

  .emotion-0::after {
    content: '';
    display: block;
    height: 0;
    width: 0;
    margin-top: -0.25795454545454544em;
  }
}

@media screen and (min-width:480px) {
  .emotion-0 {
    margin: 0 12px 0 0;
  }
}

@media screen and (min-width:960px) {
  .emotion-0 {
    margin: 0 16px 0 0;
  }
}

@media screen and (min-width:1440px) {
  .emotion-0 {
    margin: 0 16px 0 0;
  }
}

<section
    class="emotion-2"
  >
    <h5
      class="emotion-0"
    >
      this is kicker 
    </h5>
    <h4
      class="emotion-1"
    >
      test
    </h4>
  </section>
</DocumentFragment>
`;
